code/iaas/logic/src/main/java/io/cattle/platform/process/account/AccountPurge.java

```diff
@@ -49,6 +50,13 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
             deactivateThenRemove(agent, state.getData());
         }
 
+        for (Environment env : getObjectManager().children(account, Environment.class)) {
+            if (env.getRemoved() != null) {
+                continue;
+            }
+            objectProcessManager.scheduleStandardProcessAsync(StandardProcess.REMOVE, env, null);
```

Comment:
Containers' cleanup is async; so I've made services to be async as well as they are the closest to container in concept. I believe Craig did change sync to async execution for containers cleanup due to some idempotent issues (commit 7a5ed73)